src/CircFit.jl

```diff
@@ -1,7 +1,36 @@
 module CircFit
 import LsqFit: lmfit
+import Statistics: var, cov
 
-export circfit
+export circfit, Kasa
+
+@enum Method begin
+    Kasa
+end
+
+"""
+Using analytical solution of Kasa
+https://ieeexplore.ieee.org/document/1246564
+x y values are arrays
+"""
+function circfit(::Type{Val{Kasa}},x::AbstractArray,y::AbstractArray)
```

Comment:
As I commented in #1, I would drop the `Val` argument.  Just implement a single good algorithm (Kasa), and only worry about adding options for alternative algorithms in the unlikely event that there is demand.

Reply:
(It should be straightforward to derive a weighted variant of the analytical solution.)

Reply:
I opened an issue for the weighted version